Assets/Scripts/Adventurers/Adventurers.cs

```diff
@@ -57,7 +57,11 @@ public List<Adventurer> Assign(Quest quest, int adventurerCount)
         public Adventurer Assign(Quest q, string adventurerName)
         {
             Adventurer assigned = _adventurers.Find(a => a.name == adventurerName);
-            if(assigned == null) Debug.LogError("Adventurer with name " + name + " not found.");
+            if (assigned == null)
+            {
+                Debug.LogError("Adventurer with name " + name + " not found.");
+                return null;
+            }
             assigned.assignedQuest = q;
             return assigned;
```

Comment:
Was this function previously causing a null error? Danger is that this changes functionality as it now runs and returns `null` correctly.

Reply:
Yeah there was a null error if you tried to load a save file which had adventurers on a quest who weren't in the adventurers list. This won't happen naturally and can only happen if you manually edit the save file, but its a good saftey check (and I updated the method that consumes it to handle it passing back null